osu.Game/Screens/Edit/Compose/Components/SelectionHandler.cs

```diff
@@ -230,7 +234,11 @@ public void OnReleased(KeyBindingReleaseEvent<PlatformAction> e)
         /// <summary>
         /// Deselect all selected items.
         /// </summary>
-        protected void DeselectAll() => SelectedItems.Clear();
+        protected void DeselectAll()
+        {
+            SelectedItems.Clear();
+            ContextMenuContainer?.CloseMenu();
+        }
```

Comment:
One situation I'm not sure about is that pasting doesn't close the menu even though it selects the newly-pasted hitobjects.

Reply:
While pasting the object the selected object won't change, so the menu target is also the previous object that chosen. So I don't think this will cause issue

Reply:
> One situation I'm not sure about is that pasting doesn't close the menu even though it selects the newly-pasted hitobjects.

close the menu whatever the selected object changed
```diff

        internal virtual void HandleSelected(SelectionBlueprint<T> blueprint)
        {
            // there are potentially multiple SelectionHandlers active, but we only want to add items to the selected list once.
            if (!SelectedItems.Contains(blueprint.Item))
                SelectedItems.Add(blueprint.Item);

            selectedBlueprints.Add(blueprint);

+            ContextMenuContainer?.CloseMenu();
        }
```